File: C#/3 - Software design patterns/QABank/AccountFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QABank
{
    public static class AccountFactory
    {
        public static Account GetAccount(string type, int balance, string name, int overdraftLimit)
        {
            if ("Current".Equals(type, StringComparison.InvariantCultureIgnoreCase))
            {
                return new CurrentAccount(balance, name, overdraftLimit);
            }
            else if ("Savings".Equals(type, StringComparison.InvariantCultureIgnoreCase))
            {
                return new SavingsAccount(balance, name);
            }

            return null;
        }
    }
}

File: Java/3 - Software design patterns/src/main/java/com/qaicpads/AccountFactory.java
package com.qaicpads;

public class AccountFactory {
	
	public static Account getAccount(String type, int balance, String name, int overdraftLimit){
		if("Current".equalsIgnoreCase(type)) {
			return new Current(balance, name, overdraftLimit);			
		}
		else if("Savings".equalsIgnoreCase(type)) {
			return new Savings(balance, name);
		}
		
		return null;
	}

}
